src/services/componentServices.v2.js

```diff
@@ -0,0 +1,323 @@
+// TEMPORARY FILE TO STORE VERSION 2 OF SOME FUNCTIONS IN componentServices.js
+// FUNCTIONS IN THIS FILE WILL BE MOVED INTO THE ORIGINAL FILE ONCE REQUIRED
+
+const config = require('./config.json');
+const constants = config.constants;
+const facilityServices = require(config.paths.facilityServices);
+const dbContext = require('../repository/DbContext');
+const Component = require(config.paths.componentModel);
+
+/**
+ * Creates a new component.
+ * @param {Component} component must be a component object.
+ * @param {String} tempFacilityName temporary facility name which component will be assigned to, 
+ *  if it's null, then the component's facility will be assigned with facility id inside the component object
+ * @returns {Boolean} true if successful, false otherwise.
+ */
+ const addComponent = async ({ component, tempFacilityName = null }) => {
+    let resStatus = "OK";
+
+    const knex = dbContext.getKnexObject();
+
+    let newValue = convertComponentInputValue(component.value);
+
+    let newIsChild = 0;
+    let newParentId = null;
+
+    if (component.isChild) {
+        newIsChild = 1;
+        newParentId = component.parentId;
+    }
+
+    let costCalcReturn = await calculateComponentCost(component.cost, component.regionId);
+
+    if(costCalcReturn === "Cost exceeding treasury amount"){
```

Comment:
Extract this magic string pls

Reply:
This still hasn't been modified?

Reply:
Wait it should have. It's changed in the last commit. Why has this one not updated?

Reply:
Eh, really? Let me check

Reply:
Right, confirmed, resolving this :v